### DISCUSSION TITLE
dropdown menu: only close sub-menu if hovering over menu item of ancestor menu

### DISCUSSION THREAD

**Discussion by christianvoigt** [floating-ui/floating-ui#1943] [React] [answered]
Hi, I am experimenting with the [Dropdown Menu example](https://codesandbox.io/s/admiring-lamport-5wt3yg?file=/src/DropdownMenu.tsx). 

I would like to change the behaviour so that it is closer to menus in osx: an open sub-menu is not closed if I move the mouse outside of the sub-menu, except if I move the mouse over a menu item of a menu higher up in the hierarchy.

Is there an easy way to get this kind of behaviour?

**Comment by atomiks** [comment, marked as answer]
Disabling the `useHover` hook (`enabled` state) upon `mouseenter` of the submenu should achieve that behavior

**Reply by christianvoigt** [reply]
Thanks for your quick answer! 
Isn't that the same as adding `!open` to the `enabled` condition? I tried that and it did not work. 

Here is my attempt to try out your proposal: https://codesandbox.io/s/frosty-hooks-bzu8si?file=/src/DropdownMenu.tsx

But it did not work out either (maybe I misunderstood something). It seems like onMouseEnter is not called reliably if the mouse enters the sub-menu.

**Reply by atomiks** [reply]
https://codesandbox.io/s/friendly-zhukovsky-6ojikt?file=/src/DropdownMenu.tsx

It's possible for multiple menus to be open in some states here though, so probably need some logic (in the event emitter maybe?) to close any other open sibling menus.

**Reply by christianvoigt** [reply]
thanks for the example. There are also some cases in which I can move from the sub-menu to some menu items in the parent menu without closing the sub-menu. I think this happens if I move the mouse first to the left and then up, instead of moving it up and then to the left.

**Reply by atomiks** [reply]
Updated, seemed to mostly fix the other issue too

**Reply by christianvoigt** [reply]
Wow, I am really amazed by your level and speed of support! Thank you very much!
